Clippy

pitdicker opened the pull request:
Just to prevent some double work, I opened this PR.

dhardy commented on rand_core/src/lib.rs:
```diff
@@ -343,11 +343,13 @@ pub trait SeedableRng: Sized {
 
 impl<'a, R: RngCore + ?Sized> RngCore for &'a mut R {
     #[inline(always)]
+    #[cfg_attr(feature = "cargo-clippy", allow(inline_always))]
```
No, I don't want these extra lints here; if we go down that road we end up with a mess (been there). Just use `-A inline_always`.

pitdicker commented:
Removed the allow lints, and used the suggestion from `cast_lossless`. Not worth much, but doesn't hurt either.
Also changed `benches/distributions.rs` to not clash with https://github.com/rust-lang-nursery/rand/pull/344.
